### ISSUE TITLE
NSL scanner doesn't work

### ISSUE THREAD

**Issue by findryDev** [moraroy/NonSteamLaunchers-On-Steam-Deck#220]
Automatic game addition through the script is not working. Launchers install without any issues.

**Comment by Limbow000** [comment]
Same with me, didnt appear epic games when iam install with NSL v.3.8.1, please help

**Comment by moraroy** [comment]
@Limbow000 @thanks for being patient guys, currently working on the plugin to make it as best as possible, with those changes hopefully you have issues with the scanner!

**Comment by flyingthappad** [comment]
what is the use of adding launchers with separate app prefix?
are there any downsides for choosing/not choosing separate app prefix?

also, the launchers were installed but not added to the steam library, so i installed boilr and tried to add them and they were added successfully.
After that i ran the NSL script again to add YouTube, and again the shortcut to steam launcher was not added. So i ran boilr again to add it, and it did not detect/add YouTube but it also removed the already added launchers.

attaching my logs for reference.

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14488389/NonSteamLaunchers-install.log)

**Comment by hahayupgit** [comment]
Adding on to this,

NSLScanner is working well for me! (Specifically for Epic Games and Ubisoft Connect games). That being said, I can't seem to get Diablo 4 from Battle.net added through NSLScanner.

Something I'd also love to see is support for manually selecting locations of these games in the case that automatic detection doesn't work, or the games are installed in a different location besides the prefix?

Thanks so much for all of the support and great features! I'm loving the script, it's enabled me to use the Epic Games launcher with cloud-save support instead of having to go through Heroic.

**Comment by hahayupgit** [comment]
Just wanted to pop back in and say that I've managed to fix this! I think it had something to do with installing Battle.net on a separate prefix, but after doing a fresh install on the main NSL prefix and installing Diablo 4 on my microSD card, the game was detected and added! Doesn't launch automatically, but that's fine with me. I'm just happy to be able to have a separate listing I can launch into!

**Comment by Neoxon4865** [comment]
Also commenting on here that my diablo 4 installed thru the prefix selected battle.net is not being detected.

**Comment by moraroy** [comment]
@Neoxon4865 yeah its not ready for separate prefixes yet

**Comment by hydrogenicMetallium** [comment]
im just coming in to say that game scanning doesn't work for me with the plugin
Itch and GOG installed perfectly fine, however the scanner will not detect the games that are currently installed even after restarting my steam deck, not sure if im missing anything or if i need to reinstall the plugin for the scanning to work

**Comment by moraroy** [comment]
@sysmoon14

**Comment by moraroy** [comment]
@findryDev @Limbow000 @Neoxon4865 @hahayupgit has this been fixed in both the plugin and the desktop versions?

**Comment by elleon1985** [comment]
I have scanned my Epic Games flawlessly in the past, but this week I installed 2 new games and they just don't show up. I can play both games through the epic launcher itself, but they don't show up on steam like all the other previously installed games....

**Comment by moraroy** [comment]
@elleon1985 lets look at the logs

**Comment by elleon1985** [comment]
[2024-09-06 08.37.38.log](https://github.com/user-attachments/files/16904041/2024-09-06.08.37.38.log)
[2024-09-06 08.37.39.log](https://github.com/user-attachments/files/16904044/2024-09-06.08.37.39.log)
[2024-09-06 08.42.31.log](https://github.com/user-attachments/files/16904045/2024-09-06.08.42.31.log)
[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/16904054/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@elleon1985 wow you weren't kidding, I've never seen those errors before... you're using a steam deck correct?

**Comment by elleon1985** [comment]
I cannot upload the most recent scanner log as it keeps updating.....looks like this (yes, steam deck)
![20240906_090711](https://github.com/user-attachments/assets/27108284-edbc-45d0-b97e-adf3fa1f72f7)

**Comment by moraroy** [comment]
@yeah that's helpful thank you, not sure how you got those errors to be honest, did you do anything special on your deck? Or any other plug-ins you have? On pre-release or stable of decky loader?

**Comment by elleon1985** [comment]
No, I didn't do anything different with my deck.... it was working normal. But there's definitely something weird going on. Until last night, I just couldn't scan the newly installed games from Epic, but I could access the epic games launcher and play from there, and even play games from epic (that I successfully scanned in the past) straight from steam os. Today everything stopped working. Epic launcher doesn't load anymore.it goes back to steam after a few seconds... I guess I will have to go for a fresh start....thanks for your help anyways

**Comment by moraroy** [comment]
@elleon1985 no there is no need to start fresh... the issue isn't with the games or launchers, it seems to be with the plug-in itself... simply Uninstaller the plug-in and re-install using the latest version, try the one in the testing store

**Comment by elleon1985** [comment]
@moraroy just tried it, but unfortunately no luck.

**Comment by moraroy** [comment]
@elleon1985 are you on beta or preview channels of steams? Or are you on stable?

**Comment by elleon1985** [comment]
In which folder should The Epic Games Launcher be installed?

**Comment by elleon1985** [comment]
@moraroy stable channel

**Comment by moraroy** [comment]
@elleon1985 what version of decky loader? I'll do more research this weekend and see why those errors are popping up for you. I'm sorry about that

**Comment by elleon1985** [comment]
I've just left for work. I let you know the decky loader version when I get back home. thx again!

**Comment by elleon1985** [comment]
It is version v2.12.3
Nsl version 1.0.1-3130584

**Comment by elleon1985** [comment]
@moraroy could the problem be that I used the storage cleaner plug-in and cleaned all compatibility data a couple of days ago?

**Comment by moraroy** [comment]
@elleon1985 yes actually. You wiped everything , it's as if you pressed Start Fresh

**Comment by elleon1985** [comment]
@moraroy yeah, I started thinking of what could have caused any problems and that was it...  how stupid of me actually! 
I already deleted the shortcuts and installed epic again. Installing my games now.... I'll be sure to be careful when using the storage cleaner in the future.
Anyways. Thank you so much for your help and attention!

**Comment by moraroy** [comment]
@elleon1985 ah I see, no worries.. yeah Storage Cleaner works well with NSL, but only for the Shafer cache lol if you delete the compat data then yeah everything gets wiped...let me know how the version of NSL in the testing store goes

**Comment by elleon1985** [comment]
@moraroy NSL working normal again after the new install. Same with the NSL Scanner. Only problem after the new install was that one of the games I re-installed didn't support cloud saving and I lost over 30 hour of progression,  but I'm the only one to blame for that....I'm definitely staying away from the "clear all comp data" option in the future, lol
Thanks once again! Have a nice weekend!

**Comment by moraroy** [comment]
@elleon1985 ah dang sorry about that, you've given me an idea though, I'll try to implement something soon that helps for that situation I'll tag you once it's ready, I hope you have a good weekend too!

**Comment by IslandJohn** [comment]
Is this still broken? I installed it for Decky, and neither Manual scan nor Auto (and rebooting) added games I installed (as a test it was Hearthstone via Battle.net).

**Comment by CorneliousJD** [comment]
I just tried the plugin method to install B.net both as an independent prefix and under the default unified prefix and then installed Diablo 4 both ways and neither one is working unfortunately. 

Manual and auto scans plus reboots do not add it to game library? 

Am I missing something here?

Sadly spent over 200GB downloading the game multiple times today trying to get it going with the shortcut added.

**Comment by moraroy** [comment]
@CorneliousJD @IslandJohn using the latest version of both desktop or plugin, was this fixed with any of my updates? from my end things seem to be pretty stable now, just wasnt sure for battle net

**Comment by CorneliousJD** [comment]
> @CorneliousJD @IslandJohn using the latest version of both desktop or plugin, was this fixed with any of my updates? from my end things seem to be pretty stable now, just wasnt sure for battle net

yeah you had helped me via chatting on reddit, we're all good now :)

**Comment by moraroy** [comment]
@CorneliousJD cool I'll close this issue! Let me know if other issues pop up!
